src/modules/interp_met/p_lapse.cpp

```diff
@@ -0,0 +1,125 @@
+//
+// Canadian Hydrological Model - The Canadian Hydrological Model (CHM) is a novel
+// modular unstructured mesh based approach for hydrological modelling
+// Copyright (C) 2018 Christopher Marsh
+//
+// This file is part of Canadian Hydrological Model.
+//
+// Canadian Hydrological Model is free software: you can redistribute it and/or
+// modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Canadian Hydrological Model is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Canadian Hydrological Model.  If not, see
+// <http://www.gnu.org/licenses/>.
+//
+
+#include "p_lapse.hpp"
+REGISTER_MODULE_CPP(p_lapse);
+
+p_lapse::p_lapse(config_file cfg)
+        : module_base("p_lapse", parallel::data, cfg)
+{
+
+    depends_from_met("p");
+
+    provides("p");
+    provides("p_no_slope");
+
+
+    apply_cosine_correction = cfg.get("apply_cosine_correction",false);
+
+
+    LOG_DEBUG << "Successfully instantiated module " << this->ID;
+}
+void p_lapse::init(mesh& domain)
+{
+    ompException oe;
+#pragma omp parallel for
+    for (size_t i = 0; i < domain->size_faces(); i++)
+    {
+      oe.Run([&]
```

Comment:
remove the oe.Run stuff. Kevin and I found a pretty large performance regression using it in a tight loop like this. I need to come up with a better solution

Reply:
OK. Done